packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -153,6 +153,12 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
     public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId !== undefined && prevState.selectedTabId !== null) {
+            const selectedTabIdIndex = this.props.children[1].map(child => child.props.id).indexOf(this.state.selectedTabId);
+            const prevSelectedTabIdIndex = _prevProps.children[1].map(child => child.props.id).indexOf(prevState.selectedTabId);
```

Comment:
wow this is some frightening array indexing and i don't trust it one bit. shouldn't you be using `children` entirely, not the first child?
 
but actually, can you use `this.getTabChildren()` instead, like i do elsewhere in the component? you can add a param to that method to customize which `children` array it uses.